### PR TITLE
Change modal strategy

### PR DESCRIPTION
**Pull request by RandomAshes**
## Overview
This PR replaces the current modal system with one that is simpler and more accessible. It is built using the recoil app state and [react-a11y-dialog](https://github.com/KittyGiraudel/react-a11y-dialog). React-A11y-dialog is built using react portals. There is now one instance of a modal at the root of the project and it is hidden or shown based on whether the `modalContents` recoil state is populated or null.

Connects #375 

## Notes
* [Updated modal component](https://github.com/FordLabs/retroquest/pull/407/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-62a231cfd19124b0d88393fa1503499d2d3276dc7f85052eb692b1eaf22a490dR33
) using the react-a11y-dialog hook
* [Root modal instance](https://github.com/FordLabs/retroquest/pull/407/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-524d8ba8f90e17fc7f772ac33e101b7136f791d91da4e4dbf7f5a131405510aeR62)

## Testing Instructions
* Modals should look and act as they did before, with the addition of an x on the corner of the open modal for accessibility purposes.

### CODE REVIEW

File: ui/src/Common/Modal/Modal.scss [hunk 1]
```diff
@@ -17,31 +17,74 @@
...
-	@media only screen and (max-width: 610px) {
-		flex-direction: column;
-		justify-content: flex-end;
+			@include main.breakpoint-large {
+				transform: scale(1.65);
+			}
+
+			.modal-close-button {
+				margin: 1rem 1.2rem 0 0;
+				font-size: 1.5rem;
```
Comment (LowBudgetMan) [inline ui/src/Common/Modal/Modal.scss:65]: Can we make the button a bit smaller on desktop? Maybe something like
```
    margin: 1.25rem 1.5rem 0 0;
    font-size: 1rem;
```

**Review by LowBudgetMan — CHANGES_REQUESTED** [review]

File: ui/src/App/Team/Retro/ThoughtColumn/RetroItemWithAddAction/RetroItemWithAddAction.tsx [hunk 2]
```diff
@@ -14,28 +14,26 @@
...
-};
+	thoughtId: number;
+}
 
-function RetroItemModal(props: RetroItemModalProps, ref: Ref<ModalMethods>) {
-	const { type, thought, readOnly } = props;
+function RetroItemWithAddAction(props: RetroItemModalProps) {
+	const { type, thoughtId } = props;
 
+	const thought = useRecoilValue(ThoughtByIdState(thoughtId));
```
[the remark this answers is not in the document]
Comment (aclift19) [reply ui/src/App/Team/Retro/ThoughtColumn/RetroItemWithAddAction/RetroItemWithAddAction.tsx:36]: We need the action item in the modal to be able to update straight from recoil, so our options are this, or to have retroItem and the actionItem get an id and fetch the thought and action from the recoil state itself.

[the remark this answers is not in the document]
Comment (LowBudgetMan) [reply ui/src/App/Team/Retro/ThoughtColumn/RetroItemWithAddAction/RetroItemWithAddAction.tsx:36]: Alright! We should probably make the other modals behave the same way so it's more obvious.

**Review by LowBudgetMan — APPROVED** [review]

**Review by LowBudgetMan — APPROVED** [review]
lgtm

**Pull request merged**